### ISSUE TITLE
[a11y]: Overflow Menu and Modal Content is not within a landmark element

### ISSUE THREAD

**Issue by AlvinChacko** [carbon-design-system/carbon#17176]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v1.62.1

### React version

v16.12.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

All our modal and overflow menu is being flagged a s violation in the IBM 1.3.1 Info and Relationships section.

### WCAG 2.1 Violation

IBM 1.3.1 Info and Relationships

### Reproduction/example

https://react.carbondesignsystem.com/iframe.html?args=&id=components-modal--default&viewMode=story

### Steps to reproduce

Just scan the latest carbon modal and overflow menu and the violations are present for the following snippet:
```
<span class="cds--visually-hidden" role="link" tabindex="0">
```
Im not sure what this is used for or why this in the code.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@AlvinChacko What ruleset are you using? We have an open issue to update our testing pipeline to the latest:

* https://github.com/carbon-design-system/carbon/issues/17149

We usually expect to see these landmark violations within storybook because the storybook environment does not set proper landmarks and we don't configure `data-floating-menu-container`: https://github.com/carbon-design-system/carbon/issues/16946#issuecomment-2228811606 

Having a reproduction in [stackblitz](https://stackblitz.com/github/carbon-design-system/sandboxes/tree/main/react/latest/react-18-vite/?file=src%2Findex.scss,src%2FApp.jsx&preset=node=) of each of these would be ideal so we can nail down what the root issue is here.

**Comment by AlvinChacko** [comment]
Hmm I was using the the Latest Deployment Ruleset and IBM Guideline 7.2. It might be possible that I dont set the `data-floating-menu-container` because I didnt see that official documented anywhere I will try it out and then try to reproduce it.

**Comment by AlvinChacko** [comment]
Does the `data-floating-menu-container` have to go to the body tag or the container holding the modal? I tried adding it to the Modal object and it is on the parent div but the violation is still there.

**Comment by AlvinChacko** [comment]
I got it to work. Thanks for the feedback.
